### DISCUSSION TITLE
Need help with parallel output

### DISCUSSION THREAD

**Discussion by aband** [CGNS/CGNS#716] [CGNStalk]
Hi all,

I am following the example write_grid_str_parinzone.c to create a simple structured mesh in 2D.
It generated a .cgns file successfully and I could view it with paraview with no error.
Then I tried to add flow solution to the generated mesh file. 

The error occurs when cgp_open() is called. 
The error message says "cgio_open_file:not a HDF5 file - required for parallel".

I am confused which part makes the generated mesh file a hdf5 file in the given test code.

In addition, I am confused about the "Rind_t" object. I read the website page https://cgns.github.io/CGNS_docs_current/filemap/nodes.html#Rind and still can't get what it means. Can anyone give me some idea where I can find some relevant materials. 

Best,

**Comment by mennodeij** [comment]
So, CGNS has two formats: ADF and HDF5. The HDF5 support needs to be built in, using the HDF5 library, specifically using the `-DCGNS_ENABLE_HDF5=ON` flag during cmake configuring step. 

To support parallel I/O, HDF5 should also be built with MPI support (use `--enable-parallel` when building HDF5) and add `-DCGNS_ENABLE_PARALLEL=ON -DHDF5_NEED_MPI=ON` during CGNS cmake configure.

Rind_t - sorry can't help you there, haven't used it. As far as I can see, this is to support ghost cells, i.e. cells that belong to another part of the grid that is being simulated on another process/thread.

**Reply by aband** [reply]
Thank you for your reply. I am still a bit confused. 

I configured exactly as you mentioned and I make the given example with same CMake text and it succeed.

The ADF and HDF5 format are determined implicitly by CGNS with the same code?

**Reply by mennodeij** [reply]
When built with HDF5-support is enabled, this is the default. I suspect initially you may not have had HDF5 enabled in CGNS, and then it defaults to ADF.

**Reply by aband** [reply]
Thank you for the reply!

Actually, the problem happened because I forget to call cgp_close() at the end. The compile and link process are all good.
